packages/iov-tendermint-rpc/src/rpcclients/websocketclient.spec.ts

```diff
@@ -76,6 +78,42 @@ describe("WebsocketClient", () => {
     });
   });
 
+  xit("can listen to the same query twice", done => {
+    pendingWithoutTendermint();
+
+    const client = new WebsocketClient(tendermintUrl);
+
+    const newBlockHeaderQuery = "tm.event='NewBlockHeader'";
+
+    // we need two requests with unique IDs
+    const request1 = jsonRpcWith("subscribe", { query: newBlockHeaderQuery });
+    const request2 = jsonRpcWith("subscribe", { query: newBlockHeaderQuery });
+    const stream1 = client.listen(request1);
+    const stream2 = client.listen(request2);
+
+    // tslint:disable-next-line:readonly-array
+    const eventHeights: any[] = [];
+
+    const subscription = Stream.merge(stream1, stream2).subscribe({
+      error: done.fail,
+      complete: () => done.fail("subscription should not complete"),
+      next: event => {
+        // height is string or number, depending on Tendermint version. But we don't care in this case
```

Comment:
I think this is correct, but reads a bit funny...
Maybe could be a bit cleaner

Reply:
The core issue here is that we are testing an RPC client without knowing which backend this client is talking to. So we either need to make assumptions in the backend or write tests that are backend-independent.

Reply:
Sorry, it wasn't the string/number issue, but rather how it was merge and set and check....

I mean you could take both streams and do something like 

```ts
const resA = await toListPromise(streamA, 2);
const resB = await toListPromise(streamB, 2);
expect(resA.length).toEqual(2);
expect(resA).toEqual(resB);
```

Or something along those lines... gets the same checks but reads cleaner to me.

Reply:
Yeah, two aspects here:

* `toListPromise` is a very new helper that allows to write stream tests as ayncs/await instead of using `done` callbacks. This recent addition can probably cleanup dozens of tests, including this one.
* `toListPromise` on `streamA` causes a subscribe, capture, unsubscribe cycle, i.e. `resA` contains block headers number 1, 2 and `resB` contains block headers number 3, 4. Thus the Stream.merge is required.

To sum up, we can indeed do

```ts
const eventHeights = await toListPromise(Stream.merge(stream1, stream2).map(event => {
  return event.data.value.header.height;
}), 4);
expect(new Set(eventHeights).size).toEqual(2);

// that's it
```

Reply:
This is how we can rewrite it now: https://github.com/iov-one/iov-core/pull/609/commits/e1344350bd00c35112838bf37092693bd6a8f2cd?utf8=%E2%9C%93&diff=split

Reply:
Looks good